code/game/objects/items/weapons/grenades/flashbang.dm

```diff
@@ -22,45 +26,45 @@
 		B.update_icon()
 	qdel(src)
 
-/obj/item/weapon/grenade/flashbang/proc/bang(var/turf/T , var/mob/living/carbon/M)						// Added a new proc called 'bang' that takes a location and a person to be banged.
+/obj/item/weapon/grenade/flashbang/proc/bang(var/turf/T , var/mob/living/M)
 	if (locate(/obj/item/weapon/cloaking_device, M))			// Called during the loop that bangs people in lockers/containers and when banging
 		for(var/obj/item/weapon/cloaking_device/S in M)			// people in normal view.  Could theroetically be called during other explosions.
 			S.active = 0										// -- Polymorph
 			S.icon_state = "shield0"
 
-	to_chat(M, "<span class='danger'>BANG</span>")
-	playsound(get_turf(src), 'sound/effects/bang.ogg', 25, 1)
-
 //Checking for protections
-	var/eye_safety = 0
-	var/ear_safety = 0
-	if(iscarbon(M))
-		eye_safety = M.eyecheck()
-		if(ishuman(M))
-			var/mob/living/carbon/human/H = M
-			if(H.earprot())
-				ear_safety += 2
-			if(M_HULK in M.mutations)
-				ear_safety += 1
-			if(istype(M:head, /obj/item/clothing/head/helmet))
-				ear_safety += 1
+	var/eye_safety = M.eyecheck()
+	var/ear_safety = M.earprot() * 2 //some arbitrary measurement of ear protection, I guess? doesn't even matter if it goes above 1
+
+	if(ishuman(M))
+		if(istype(M:head, /obj/item/clothing/head/helmet))
```

Comment:
The bizarre thing is that the old code cast to human and then still used a colon.
Make this a cast.